src/managers/WebhookManager.js

```diff
@@ -34,29 +40,157 @@ class WebhookManager {
+          SPARK_PUBLISHED_AT: event.publishedAt,
+        };
+
+        let eventDataVariables = {};
+        try {
+          eventDataVariables = JSON.parse(event.data);
+        } catch (error) {
+          eventDataVariables = {};
+        }
+
+
+        const webhookVariablesObject = webhook.noDefaults
+          ? eventDataVariables
+          : {
+            ...defaultWebhookVariables,
+            ...eventDataVariables,
+          };
+
+        const requestJSON = webhook.json && JSON.parse(
+            hogan
+              .compile(JSON.stringify(webhook.json))
+              .render(webhookVariablesObject),
+          );
+
+        const requestFormData = webhook.form && JSON.parse(
+            hogan
+              .compile(JSON.stringify(webhook.form))
+              .render(webhookVariablesObject),
+          );
+
+        const requestUrl = hogan
+          .compile(webhook.url)
+          .render(webhookVariablesObject);
+
+        const requestQuery = webhook.query && JSON.parse(
+            hogan
+              .compile(JSON.stringify(webhook.query))
+              .render(webhookVariablesObject),
+          );
+
+        const responseTopic = webhook.responseTopic && hogan
+            .compile(webhook.responseTopic)
+            .render(webhookVariablesObject);
+
+        const errorResponseTopic = webhook.errorResponseTopic && hogan
+            .compile(webhook.responseTopic)
+            .render(webhookVariablesObject) || `hook-error/${event.name}`;
+
+        const responseHandler = (
+          error: ?Error,
+          response: http$IncomingMessage,
+          responseBody: string | Buffer | Object,
+        ) => {
           if (error) {
-            // todo block the webhook calls after some amount of fails
-            // on 1 min or so..
-            // todo responseTemplates
+            this._incrementWebhookErrorCounter(webhook.id);
+
+            this._eventPublisher.publish({
+              data: error.message,
+              name: errorResponseTopic,
+              userID: event.userID,
+            });
+
+            return;
             throw error;
           }
+
+          this._resetWebhookErrorCounter(webhook.id);
+
+          this._eventPublisher.publish({
+            name: `hook-sent/${event.name}`,
+            userID: event.userID,
+          });
+
+          const responseTemplate = webhook.responseTemplate && hogan
+              .compile(webhook.responseTemplate)
+              .render(responseBody);
+
+          if (responseTopic) {
+            this._eventPublisher.publish({
+              data: webhook.responseTemplate && responseTemplate,
+              name: responseTopic,
+              userID: event.userID,
+            });
+          }
         };
 
-        // todo request <-> webhooks options
-        request({
-          body: webhook.json,
-          formData: webhook.form,
+
+        const requestOptions = {
+          body: requestJSON || event.data,
```

Comment:
@AntonPuko why do you need this? it throws an error for me:
`Caught exception: Error: write after end Error: write after end
    at ClientRequest.OutgoingMessage.write (_http_outgoing.js:439:15)`

if i comment the body request options it works